src/App.tsx

```diff
@@ -1,22 +1,57 @@
 import classnames from 'classnames'
 import React from 'react'
+import _ from 'lodash/fp'
 import {SplashScreen} from './SplashScreen'
-import {BackendState} from './common/backend-state'
+import {
+  BackendState,
+  defaultBackendData,
+  migrationsForBackendData,
+  StoreBackendData,
+} from './common/backend-state'
 import {rendererLog} from './common/logger'
-import {createPersistentStore} from './common/store'
-import {WalletState} from './common/wallet-state'
+import {createPersistentStore, Store} from './common/store'
+import {WalletState, defaultWalletData, StoreWalletData} from './common/wallet-state'
 import {config} from './config/renderer'
 import {Router} from './layout/Router'
 import {Sidebar} from './layout/Sidebar'
 import {RouterState} from './router-state'
-import {SettingsState} from './settings-state'
-import {TokensState} from './tokens/tokens-state'
+import {SettingsState, defaultSettingsData, StoreSettingsData} from './settings-state'
+import {TokensState, defaultTokensData, StoreTokensData} from './tokens/tokens-state'
 import {TransactionHistoryService} from './wallets/history'
 import {createWeb3} from './web3'
 import './App.scss'
 
 const web3 = createWeb3(config.rpcAddress)
-const store = createPersistentStore()
+
+// TODO: move migrations and default data related code after restructuring store/state related code
```

Comment:
Please, create a Jira task with the details of the next step and add the issue number here.

Reply:
Added a jira task and issue number. However, I still need to think about what would be the optimal way to restructure it :)